content/en/blog/2020/workload-entry/index.md

```diff
@@ -0,0 +1,100 @@
+publishdate: 2020-05-14
+attribution: "Cynthia Coan (Tetrate), Shriram Rajagopalan (Tetrate), Tia Louden (Tetrate), John Howard (Google), Sven Mawson (Google)"
+keywords: [vm,workloadentry,migration,'1.6',baremetal,serviceentry,discovery]
+---
+
+## Introducing Workload Entries: Bridging Kubernetes and VMs
+
+Historically, Istio has provided great experience to workloads that run on Kubernetes, but a less than smooth one for other types of workloads, such as Virtual Machines (VM) and Bare Metal. The gaps included the inability to declaratively specify the properties of a sidecar on a VM, inability to properly respond to the lifecycle changes of the workload (e.g., booting to not ready to ready, or health checks), and cumbersome DNS workarounds as the workloads are migrated into Kubernetes to name a few.
+
+Istio 1.6 has introduced a few changes in how you manage non-Kubernetes workloads, driven by a desire to make it easier to gain Istio's benefits for use cases beyond containers, such as running traditional databases on a platform outside of Kubernetes, or adopting Istio's features for existing applications without rewriting them.
+
+### Background
+
+Prior to 1.6, non-containerized workloads were being treated as an IP address in a Service Entry, even if they had a sidecar. Istio lacked a first class abstraction for non-containerized workloads, similar to how Kubernetes treats Pods as the fundamental unit of compute - a named object that served as the collection point for all things related to a workload - name, labels, security properties, lifecycle status events, etc. Enter Workload Entry.
+
+Consider the following Service Entry describing a service implemented by a few tens of VMs with IP addresses:
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: svc1
+spec:
+  hosts:
+  - svc1.internal.com
+  ports:
+  - number: 80
+    name: http
+    protocol: HTTP
+  resolution: STATIC
+  endpoints:
+  - address: 1.1.1.1
+  - address: 2.2.2.2
+  ....
+{{< /text >}}
+
+If you wanted to migrate this service into Kubernetes in an active-active manner - i.e. launch a bunch of pods, send a portion of the traffic to the pods over Istio mutual TLS (mTLS) and send the rest to the VMs without sidecars. How would you do it? You would have needed to use a combination of a Kubernetes service, a virtual service, and a destination rule to achieve the behavior. Now, let's say you decided to add sidecars to these VMs, one by one, such that you want only the traffic to the VMs with sidecars to use Istio mTLS. If any other Service Entry happens to select the same VM for its service, things start to get very complicated and error prone.
+
+The source of these complications is that Istio lacked a first class definition of a non-containerized workload, whose properties can be described independently of the service(s) it binds to.
+
+{{< image width="75%"
```

Comment:
Could @rcaballeromx have a go at this SVG?

Reply:
@craigbox is there perhaps something we could fix with these SVGs ourselves to be more in line? Perhaps that way it can be done on our side quicker? I imagine it's more work for someone else to take a whole new whack at it.

Reply:
PTAL at https://istio.io/about/contribute/diagrams/ and see if that helps any.

Reply:
@SecurityInsanity For a blog, your diagrams look fine to me, just a little small with blank space around them. See if you can fix that. Take a look at the preview: https://deploy-preview-7321--preliminary-istio.netlify.app/blog/2020/workload-entry/#introducing-workload-entries-bridging-kubernetes-and-vms.

Also, in the second diagram, the box labeled `Service` is a little confusing. In the first diagram the box is labeled `K8S Service`, which makes sense. In the second diagram, however, it's not a `K8S Service`, but rather an `istio ServiceEntry`, so labeling it as just `Service` seems to confuse the picture.

Reply:
Here is an exmaple of a diagram done in google draw. I drew this in 12 minutes. Google draw can be exported directly toe SVG and then (reused) by others, whereas a PNG cannot. This is one of the reasons we prefer SVG in the docs, https://docs.google.com/drawings/d/1xhAHscnZB3vJiVvuhi0K0DPqQmE9Pk0szVAm5vW0pB0/edit?usp=sharing

Cheers,
-steve